src/Middleware/OutputCaching/src/OutputCacheEntryFormatter.cs

```diff
@@ -78,7 +79,21 @@ public static async ValueTask StoreAsync(string key, OutputCacheEntry value, Tim
 
         Serialize(bufferStream, formatterEntry);
 
-        await store.SetAsync(key, bufferStream.ToArray(), value.Tags ?? Array.Empty<string>(), duration, cancellationToken);
+        if (!bufferStream.TryGetBuffer(out var segment))
+        {
+            segment = bufferStream.ToArray();
+        }
+
+        var payload = new ReadOnlySequence<byte>(segment.Array!, segment.Offset, segment.Count);
+        if (store is IOutputCacheBufferStore bufferStore)
+        {
+            await bufferStore.SetAsync(key, payload, value.Tags, duration, cancellationToken);
```

Comment:
Is this PR also going to update the middleware to use `TryGetAsync` when it's available?

Reply:
That will go into the perf PR which already has the relevant buffer work in place to work there. If I can merge both for preview 6: great, but the critical path is getting it *functionally* usable - I suspect we'll have the perf rebased in at most 2 days, but maybe not for this preview. Basically, the perf PR already does all the things we need to use that API.